server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -406,33 +440,20 @@ public void removeBan(TaskId parentTaskId) {
      * @param onChildTasksCompleted called when all child tasks are completed or failed
      * @return the set of current nodes that have outstanding child tasks
      */
-    public Collection<DiscoveryNode> startBanOnChildrenNodes(long taskId, Runnable onChildTasksCompleted) {
+    public Map<String, List<DiscoveryNode>> startBanOnChildrenNodes(long taskId, Runnable onChildTasksCompleted) {
         final CancellableTaskHolder holder = cancellableTasks.get(taskId);
         if (holder != null) {
             return holder.startBan(onChildTasksCompleted);
         } else {
             onChildTasksCompleted.run();
-            return Collections.emptySet();
+            return Collections.emptyMap();
         }
     }
 
     @Override
     public void applyClusterState(ClusterChangedEvent event) {
         lastDiscoveryNodes = event.state().getNodes();
         if (event.nodesRemoved()) {
-            synchronized (banedParents) {
-                lastDiscoveryNodes = event.state().getNodes();
```

Comment:
do we still need some of this for BWC?

Reply:
Good catch. I restored it in https://github.com/elastic/elasticsearch/pull/55779/commits/617bc358e5b82ad48d1e0f6986259fb837546dc2.